src/python/tritonfrontend/CMakeLists.txt

```diff
@@ -0,0 +1,187 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required(VERSION 3.18)
+
+# ================= Ensures Package is Structured Properly ==================
+file(COPY __init__.py DESTINATION .)
+file(COPY py.typed DESTINATION .)
+file(COPY _c/__init__.py DESTINATION ./_c/.)
+file(COPY _c/__init__.pyi DESTINATION ./_c/.)
+
+file(GLOB PYTHON_MODULE_FILES ./_api/*.py)
+file(COPY ${PYTHON_MODULE_FILES} DESTINATION ./_api/.)
+# ================================= END =====================================
+
+
+# =================== Downloading and Installing pybind11 ===================
+include(FetchContent)
+
+FetchContent_Declare(
+    pybind11
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG v2.13.1
+    GIT_SHALLOW ON
+)
+
+FetchContent_MakeAvailable(pybind11)
+# ================================= END =====================================
+
+# ================== Collect the Dependencies ===============================
+set(
+  PYTHON_FRONTEND_BINDING_DEPS
+  ../../shared_memory_manager.h
+  ../../shared_memory_manager.cc
+  ../../data_compressor.h
+  ../../common.h
+  ../../common.cc
+  ../../restricted_features.h
+  ../../tracer.h
+  ../../classification.cc
+)
+
+if(${TRITON_ENABLE_TRACING})
+  list(APPEND PYTHON_FRONTEND_BINDING_DEPS
+      ../../tacer.cc
+    )
+endif()
```

Comment:
Wondering may be we can use [cmake generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) in this cenario.
```suggestion
  ../../tracer.h
  $<$<BOOL:${TRITON_ENABLE_TRACING}>: ../../tacer.cc>
  ../../classification.cc
)
```

Reply:
When I attempt to build with the generator expression, if cmake is run with `-DTRITON_ENABLE_TRACING:BOOL=ON`, I get the following error during the build process:
![image](https://github.com/user-attachments/assets/bf587129-d408-4f94-8373-97a4123df661)
Does this mean that generator expression is not getting evaluated? Or that this is the result of the evaluation?

Reply:
Can you share snipped you post in your CMake

Reply:
I took another stab at it this morning and realized that I added an extra pair of braces in the expression, which led to the error. It is working now. Thank you for the suggestion.